src/TestSuite/TestCase.php

```diff
@@ -249,9 +250,56 @@ protected function tearDown(): void
         $this->_configure = [];
         $this->_tableLocator = null;
 
-        if ($this->fixtureManager) {
-            $this->fixtureManager->teardownTest($this);
-            $this->fixtureManager = null;
+        $this->teardownFixtures();
+    }
+
+    /**
+     * Initialized and loads any use fixtures.
+     *
+     * @return void
+     */
+    protected function setupFixtures(): void
+    {
+        $fixtureNames = $this->getFixtures();
+        if (empty($fixtureNames)) {
+            return;
+        }
+
+        if (!isset(static::$schemaStrategies[$this->schemaStrategyClass])) {
+            $schemaStrategy = new $this->schemaStrategyClass();
+            if (!$schemaStrategy instanceof SchemaStrategyInterface) {
+                throw new RuntimeException(
+                    "Schema strategy `{$this->schemaStrategyClass}` does not implement `SchemaStrategyInterface`."
+                );
+            }
+            $schemaStrategy->load();
+            static::$schemaStrategies[$this->schemaStrategyClass] = $schemaStrategy;
+        }
+
+        static::$fixtureStrategy = new $this->fixtureStrategyClass();
+        if (!static::$fixtureStrategy instanceof FixtureStrategyInterface) {
+            throw new RuntimeException(
+                "Fixture strategy `{$this->fixtureStrategyClass}` does not implement `FixtureStrategyInterface`."
+            );
+        }
+        static::$fixtureStrategy->setupTest($fixtureNames);
```

Comment:
I like this but I think we might need to also have an 'aggregate' fixture strategy. An application that needs to straddle multiple datasources would need to load/reset state on all of those storage systems which is harder to do now.